components/purchase_modal/purchase_modal.tsx

```diff
@@ -609,6 +607,16 @@ class PurchaseModal extends React.PureComponent<Props, State> {
         );
     }
 
+    selectedProduct = () => {
+        // If the subscription is not past 90 days delinquent, we don't need to upgrade them as they're still on the correct plan
+        if (this.props.isDelinquencyModal && !this.props.isCloudDelinquencyGreaterThan90Days) {
+            return findProductInDictionary(this.props.products, this.props.productId);
+        }
+
+        // If not delinquent, or they've been downgraded (ie, passed 90 days), the modal functions as normal
+        return this.state.selectedProduct;
+    }
+
```

Comment:
This works but we could also add that condition where the `selectedProduct` is initially set [here](https://github.com/mattermost/mattermost-webapp/blob/8f81a8278f3d3eddd4ff087794a5525888d591e0/components/purchase_modal/purchase_modal.tsx#L166-L174 ) This way we leave all the logic of selecting the current product in the same place and don't have issues of in the future accessing `this.state.selectedProduct` incorrectly